src/libraries/Native/Unix/System.Native/pal_interfaceaddresses.c

```diff
@@ -373,7 +373,7 @@ int32_t SystemNative_GetNetworkInterfaces(int32_t * interfaceCount, NetworkInter
                         ecmd.cmd = ETHTOOL_GSET;
                         if (ioctl(socketfd, SIOCETHTOOL, &ifr) == 0)
                         {
-                            nii->Speed = (int)ethtool_cmd_speed(&ecmd);
+                            nii->Speed = (int64_t)ethtool_cmd_speed(&ecmd);
```

Comment:
AFAICT, ethtool_cmd_speed returns an [unsigned 32-bit value](https://github.com/torvalds/linux/blob/bd2463ac7d7ec51d432f23bf0e893fb371a908cd/include/uapi/linux/ethtool.h#L123). Why does it need to be cast to a 64-bit value here? Just because the NetworkInterface.Speed property is a long? That cast happens later implicitly in LinuxNetworkInterface, right?

Reply:
Never mind. I read the issue more carefully. The problem is just below the fold on [line 380](https://github.com/dotnet/runtime/blob/48080fa967e85a4aa162281a950358eea430e5f4/src/libraries/Native/Unix/System.Native/pal_interfaceaddresses.c#L380) (`nii->Speed *= 1000000;`).

Reply:
There's a multiply a few lines later that is overflowing.